cfme/tests/configure/test_vmware_console_settings.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.configure.configuration import VMwareConsoleSupport
+from utils.appliance.implementations.ui import navigate_to
+from utils.version import current_version
+
+
+@pytest.mark.tier(3)
+@pytest.mark.sauce
+@pytest.mark.uncollectif(lambda: current_version() < '5.8')
+def test_vmware_console_support(appliance):
+    """Tests that the VMware Console Support setting may be changed."""
+    navigate_to(appliance.server, 'Server')
+
+    console_type_widget = VMwareConsoleSupport.vmware_console_form.console_type
+    old_vm_console_type = console_type_widget.first_selected_option_text
+    assert old_vm_console_type, "The default VMware console type should not be empty"
+
+    for new_vm_console_type in VMwareConsoleSupport.CONSOLE_TYPES:
+        console_settings = VMwareConsoleSupport(
```

Comment:
So I tried to use the context manager (with update), but it just didn't work with this object.   Turns out that if I do something like:

        vmware_console_settings = VMwareConsoleSupport(
            appliance=appliance,
            console_type=new_vm_console_type
        )
        ...
        vmware_console_settings.console_type
        vmware_console_settings.update()

The update does not work.   It's like it doesn't actually try to change the settings.   The
VMwareConsoleSupport class was setup to mimic other classes in cfme/configure/configuration/__init__.py, so its doing the same things as them.   Which is to say this whole file needs revisiting at some point in time.   But either way because changing the console_type attribute after object instantiation and then calling update does not work, the context manager does not work.    This also affects creating a fixture and passing around the same VMwareConsoleType object.